### ISSUE TITLE
sequelize.transaction() convention proposal

### ISSUE THREAD

**Issue by cusspvz** [sequelize/sequelize#2717]
I've been using `sequelize.transaction()` for a while, and since it is returning a `Promise` instance, i've been coding like this:

```js
var Ms = sequelize.models;

sequelize.transaction()
  // Create a context for promise chain
  .bind({})
  // Bind transaction to it
  .then(function ( t ) { this.transaction = t; })
  
  // now i grab transaction from promise chain
  .then(function () {
    return Ms.SomeModel.find({ /* ... */ }, this)
  })
  .then(function ( instance ) {
    if ( instance ) return;

    // If i need to return a promise, on this then ...
    var someinstance = Ms.SomeModel.build();

    return someinstance.save( this )
      // I just bind current context
        .bind( this )
        .then(function () {
          return someinstance.someMethodThatNeedsOptionsWithTransaction( this )
        });
  })
  .finally(function () {
    return this.transaction.commit();
  });

```

Since #1627 implementation isn't decided yet, what do you think about making `sequelize.transaction` method to return a context with `transaction` within?

**Comment by cusspvz** [comment]
Seems that this is solved with #2810 right ? @mickhansen @janmeier

**Comment by janmeier** [comment]
@cusspvz Yep
